converged_sound_open_fw_modules_development/modules/aca_module/aca_config.h

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: BSD-3-Clause
```

Comment:
this should be in src/audio/aca

Reply:
> this should be in src/audio/aca

@lgirdwood we already had a little mess in src/audio where we mix SOF core built-in modules and optional 3rd party SOF module examples like smart_amp, lets not add loadable modules on top of that.

I strongly recommend to split the loadable module examples into separate folder. The existing SOF modules examples should be in the future made as loadable as well and moved to new folder.

Reply:
@lgirdwood 
Are you referring only to headers or to source files  with makefiles also?  If we move there at the moment source files with meakefiles we will have two different building systems in ```src/audio```.

Reply:
We are aligning **all** components/modules and the codec adapter to all use the same interface so it really makes sense to try and put them in the right place from day 1 otherwise it becomes difficult to port code between versions (this also includes Kconfig). 
There is also no mess in mixing the opensource components/modules together with open source glue for the closed source

Reply:
> We are aligning **all** components/modules and the codec adapter to all use the same interface so it really makes sense to try and put them in the right place from day 1 otherwise it becomes difficult to port code between versions (this also includes Kconfig).
> There is also no mess in mixing the opensource components/modules together with open source glue for the closed source

The codec adapter and loadable modules have different constraints (& APIs) then built-in FW modules. That is why I would recommend to use this opportunity and create dedicated folder for loadable libraries code and their infrastructure. 

Overall with introduction of loadable libraries we will no longer statically compile modules into FW.

Reply:
Sorry I'm not following any of this. So I think I'm reading here there will be new processing components being added but they use a different API ?
@lgirdwood I also think you are saying that the processing APIs will be "aligned" to a single API that all processing will use ?
If these statements are correct it seems obvious that we need to align the processing API **first** and then upstream the new code otherwise we have unusable code that cant be built. Unless I'm missing something here.

Reply:
@jgunn0262 yes, we are aligning the processing APIs. The intention is to have a C and a C++ API that exports the same methods and control structures. Our intention with this and other similar PRs is to provide a buildable preview for the new work.

Reply:
@lgirdwood thanks, but this code has **nothing** in common with the upstream code today. I would suggest we follow normal process and either put this under a "staging" directory e.g. ```src/staging/this_code``` or in a staging repo until all the dependencies are merged upstream.

Reply:
It seems another way would be to create a new ```sof_modules_staging``` repo and copy all this code "as-is" for preview and reference. This would the give the visibility whilst Intel worked on upstreaming incrementally, i.e. after the component/module API is aligned upstream